backend/asmgen.ml

```diff
@@ -433,13 +433,16 @@ let compile_unit ~output_prefix ~asm_filename ~keep_asm ~obj_filename ~may_reduc
 
 let build_dwarf ~asm_directives:(module Asm_directives : Asm_targets.Asm_directives_intf.S) sourcefile =
   let unit_name =
-    Compilation_unit.get_persistent_ident (Compilation_unit.get_current_exn ())
+    Compilation_unit.get_current_exn ()
+    |> Symbol.for_compilation_unit
+    |> Symbol.linkage_name
+    |> Ident.create_persistent
   in
   let code_begin =
-    Compilenv.make_symbol (Some "code_begin") |> Asm_targets.Asm_symbol.create
+    Cmm_helpers.make_symbol "code_begin" |> Asm_targets.Asm_symbol.create
   in
   let code_end =
-    Compilenv.make_symbol (Some "code_end") |> Asm_targets.Asm_symbol.create
+    Cmm_helpers.make_symbol "code_end" |> Asm_targets.Asm_symbol.create
   in
   Dwarf.create
```

Comment:
`Dwarf.create` should be fixed to take `Linkage_name.t`, probably.  We shouldn't be translating back to `Ident` at this stage.

Reply:
Yeah, as far as I can tell it's only ever used in `Dwarf_attribute_helpers.create_ocaml_unit_name`, where it's immediately passed to `Ident.name`. I'll leave a CR (starting to worry about mission creep).